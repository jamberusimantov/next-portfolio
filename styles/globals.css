html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
    opacity: 1;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.submitBtn {
    width: 100%;
    max-width: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: .5rem;
}

.submitBtn p {
    flex: 1;
    margin: 0;
    font-size: large;
    padding: 0 !important;
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

@media(min-width:580px) {
    .submitBtn {
        margin: auto;
    }
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}