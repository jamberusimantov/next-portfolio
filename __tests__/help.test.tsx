/**
 * @jest-environment jsdom
 */
 import React, { ReactElement, JSXElementConstructor } from "react";
 import { describe, expect, test } from '@jest/globals';
 import { render, act } from '@testing-library/react';
 import Help from '../pages/help';
 
 
 describe('Page: Help', () => {
    test('use jsdom in this test file', () => {
        expect(document.createElement('div')).not.toBeNull();
    });
    test('renders correctly', async () => {
        let el: HTMLElement = document.createElement("div");
        await act(async () => {
            const getLayout = Help.getLayout ?? (page => page);
            const {container} = await render(getLayout(<Help />));
            el = container;
        });
        expect(el).toMatchSnapshot();
    });
});