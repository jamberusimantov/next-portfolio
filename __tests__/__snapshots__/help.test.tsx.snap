// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Help renders correctly 1`] = `
<div>
  <div
    class="layout"
  >
    <header
      class="header"
    >
      <div
        class="headerTop"
      >
        <div
          class="headerTopContainer"
        >
          <div
            class="logo"
          >
            <div
              class="logoContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="name"
            >
              Siman Tov Jamberu
            </span>
          </div>
        </div>
      </div>
    </header>
    <main>
      <div
        class="page"
      >
        <div
          class="pageContainer"
        >
          <div
            class="navSection"
          >
            <div
              class="navSectionContainer"
            >
              <div
                class="nav"
              >
                <div
                  class="breadcrumbs"
                >
                  <a
                    class="white"
                    href="/"
                    title="Homepage"
                  >
                    דף הבית
                  </a>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="disabled"
                    href="/help"
                  >
                    עזרה
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="help"
            >
              <h1>
                עזרה
              </h1>
              <p>
                portfoio-simantov.herokuapp.com מספק מידע מקצועי על סימן טוב ג'מברו. ניתן ואף רצוי לשאול שאלות וליצור קשר בכל נושא בקשר לאתר או אליי.
              </p>
              <h2>
                מסמכים ב-portfoio-simantov.herokuapp.com
              </h2>
              <p>
                portfoio-simantov.herokuapp.com מגיש מסמכים בתצורת PDF. על מנת להציג ולהדפיס יש צורך בתוכנה מתאימה. תוכנות מסוג זה זמינות למגוון מערכות הפעלה כגון: Windows, Mac OS, Linux וניתנות בדרך כלל להורדה ללא עלות. לדוגמא Adobe Reader.
              </p>
              <h2>
                שאלות ומשוב ב-portfoio-simantov.herokuapp.com
              </h2>
              <p>
                בכוונתי לשמור על מידע מקיף, מדוייק, מעודכן וידידותי למשתמש ככל שיעלה בידיי. האתר מנוהל  על ידי והוא בפיתוח מתמשך. אם צצה לך שאלה, משוב או הצעה לגביי האתר, אשמח לשמוע וניתן לפנות בעזרת טופס 
                <a
                  class="link"
                  href="/contactByEmail"
                  title="שלח דוא"ל"
                >
                  שלח דוא"ל
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="footer"
    >
      <div
        class="footerTop"
      >
        <div
          class="footerTopContainer"
        >
          <div
            class="footerLinks"
          >
            <div
              class="serviceLinks"
            >
              <h3>
                שירותים
              </h3>
              <ul>
                <li>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="white"
                    href="/contact"
                    title="Contact"
                  >
                    צור קשר
                  </a>
                </li>
                <li>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="white"
                    href="/help"
                    title="Help"
                  >
                    עזרה
                  </a>
                </li>
                <li>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="white"
                    href="/archive"
                    title="Archive"
                  >
                    ארכיון
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footerBottom"
      >
        <div
          class="footerBottomContainer"
        >
          <div
            class="footerLang"
          >
            <div
              class="lang"
            >
              <button
                class=" button"
                type="button"
              >
                <p>
                  English
                </p>
              </button>
            </div>
            <div
              class="lang"
            >
              <button
                class="currentLang button"
                type="button"
              >
                <p>
                  עברית
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
